src/Raven.Server/RavenServer.cs

```diff
@@ -1332,6 +1315,80 @@ private void ListenToNewTcpConnection(TcpListener listener)
                 }
             });
         }
+
+        private async Task<TcpClient> AcceptTcpClientAsync(TcpListener listener)
+        {
+            var backoffSecondsDelay = 0;
+            while (true)
+            {
+                try
+                {
+                    return await listener.AcceptTcpClientAsync();
+                }
+                catch (ObjectDisposedException)
+                {
+                    // shutting down
+                    return null;
+                }
+                catch (Exception e)
+                {
+                    if (backoffSecondsDelay == 0)
+                    {
+                        if (_tcpLogger.IsInfoEnabled)
+                        {
+                            _tcpLogger.Info($"Failed to accept new tcp connection, will retry now", e);
+                        }
+
+                        backoffSecondsDelay = 1;
+                        continue;
+                    }
+
+                    if (_tcpLogger.IsOperationsEnabled)
+                    {
+                        _tcpLogger.Operations($"Failed to accept new tcp connection again, will wait {backoffSecondsDelay} seconds before retrying", e);
+                    }
+
+                    try
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(backoffSecondsDelay), ServerStore.ServerShutdown);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // shutting down
+                        return null;
+                    }
+
+                    if (listener.Server.Connected)
+                    {
+                        backoffSecondsDelay = Math.Min(backoffSecondsDelay * 2, 60);
+                        continue;
+                    }
+
+                    var msg = $"The socket on {listener.LocalEndpoint} is no longer connected.";
+
+                    if (_tcpLogger.IsOperationsEnabled)
+                    {
+                        _tcpLogger.Operations(msg, e);
+                    }
+
+                    var alert = AlertRaised.Create(
+                        null,
+                        msg,
+                        $"There are four threads listening for a new TCP connections, if any of them is down it is a severe error that can affect performance, when all of them are down it will cause TCP connectivity issues.{Environment.NewLine}" +
```

Comment:
This is describing implementation details. Let's just explain what is going on.
Unable to accept connections from TCP and the listen socket was rudely disconnected or something like that

Reply:
I though what should I write to avoid support calls :)

What do you suggest? should what you proposed should be enough `Unable to accept connections from TCP and the listen socket was rudely disconnected` ?